### PR TITLE
logtail: cap the buffer size in encodeText

### PR DESCRIPTION
**Pull request by josharian**
This started as an attempt to placate GitHub's code scanner,
but it's also probably generally a good idea.

Signed-off-by: Josh Bleecher Snyder <josh@tailscale.com>

### CODE REVIEW

File: logtail/logtail.go [hunk 1]
```diff
@@ -431,6 +431,21 @@ func (l *Logger) encodeText(buf []byte, skipClientTime bool) []byte {
 	// For now just factor in a dozen.
 	overhead += 12
 
+	// Put a sanity cap on buf's size.
+	max := 16384
```
Comment (bradfitz) [inline logtail/logtail.go:435]: 16 << 10 looks less mysterious

After 4096 I start forgetting my powers of two in decimal

File: logtail/logtail.go [hunk 2]
```diff
@@ -469,14 +484,10 @@ func (l *Logger) encodeText(buf []byte, skipClientTime bool) []byte {
...
-		if l.lowMem && i > 254 {
-			// TODO: this can break a UTF-8 character
-			// mid-encoding.  We don't tend to log
-			// non-ASCII stuff ourselves, but e.g. client
-			// names might be.
-			b = append(b, "…"...)
-			break
-		}
+	}
+	if nTruncated > 0 {
```
Comment (bradfitz) [inline logtail/logtail.go:488]: testable?

File: logtail/logtail.go (hunk 1 above)
Comment (josharian) [reply logtail/logtail.go]: I feel the opposite, but sure, done.

File: logtail/logtail.go (hunk 2 above)
Comment (josharian) [reply logtail/logtail.go:488]: tested

**Review by bradfitz — APPROVED** [review]

**Review by crawshaw — APPROVED** [review]

**Pull request merged**